Framework/Kernel/inc/MantidKernel/MaskedProperty.h

```diff
@@ -27,7 +27,7 @@ namespace Kernel {
 template <typename TYPE = std::string> class MaskedProperty : public Kernel::PropertyWithValue<TYPE> {
 public:
   /// Constructor with a validator
-  MaskedProperty(const std::string &name, TYPE defaultvalue,
+  MaskedProperty(const std::string &name, const TYPE &defaultvalue,
```

Comment:
I am getting cppcheck `passedByValue` warnings for these constructors and the `LoadStl` constructors when I change it to "pass by value". I think its because I needed to "pass by reference" in these constructors, and then the data will get "passed by value" to the parent constructors here:
https://github.com/mantidproject/mantid/blob/d529618f2eed7c53a7c1b90bd0afb349437c514e/Framework/Kernel/src/MaskedProperty.cpp#L26

Is this the right thing to do?

Reply:
Oh. No we should need to pass by reference anywhere. Do we need to `move` here too:
```cpp
Kernel::PropertyWithValue<TYPE>(std::move(name), std::move(defaultvalue), std::move(validator), direction)
```
The move needs to happen with each value and not just once.

Reply:
Ah ok, thats worked - thanks